presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java

```diff
@@ -275,38 +279,82 @@ private void doTestHiveCoercion(HiveTableDefinition tableDefinition)
+                                asMap(2, asMap("ti2bi", -3L, "int2bi", 2323L, "float2double", 0.5, "add", null)),
+                                1),
+                        row(
+                                1,
+                                -2,
+                                null,
+                                -100,
+                                101L,
+                                -2323L,
+                                "-12345",
+                                -1.5,
+                                asMap("keep", null, "ti2si", (short) 1, "si2int", -100, "int2bi", -2323L, "bi2vc", "-12345"),
+                                ImmutableList.of(asMap("ti2int", -2, "si2bi", 101L, "bi2vc", "-12345")),
+                                ImmutableMap.of(-2, asMap("ti2bi", null, "int2bi", -2323L, "float2double", -1.5, "add", null)),
+                                1));
+        }
+        else if (usingTeradataJdbcDriver(connection)) {
+            expectedRows = ImmutableList.of(
+                    row(
+                            -1,
+                            2,
+                            -3L,
+                            100,
+                            -101L,
+                            2323L,
+                            "12345",
+                            0.5,
+                            "[\"as is\",-1,100,2323,\"12345\"]",
+                            "[[2,-101,\"12345\"]]",
+                            "{\"2\":[-3,2323,0.5,null]}",
+                            1),
+                    row(
+                            1,
+                            -2,
+                            null,
+                            -100,
+                            101L,
+                            -2323L,
+                            "-12345",
+                            -1.5,
+                            "[null,1,-100,-2323,\"-12345\"]",
+                            "[[-2,101,\"-12345\"]]",
+                            "{\"-2\":[null,-2323,-1.5,null]}",
+                            1));
+        }
+        else {
+            throw new IllegalStateException();
+        }
         // test primitive values
         assertThat(queryResult.project(1, 2, 3, 4, 5, 6, 7, 8, 12)).containsOnly(project(expectedRows, 1, 2, 3, 4, 5, 6, 7, 8, 12));
         // test structural values (tempto can't handle map and row)
         assertEqualsIgnoreOrder(queryResult.column(9), column(expectedRows, 9), "row_to_row field is not equal");
-        assertEqualsIgnoreOrder(extract(queryResult.column(10)), column(expectedRows, 10), "list_to_list field is not equal");
+        if (usingPrestoJdbcDriver(connection)) {
+            assertEqualsIgnoreOrder(extract(queryResult.column(10)), column(expectedRows, 10), "list_to_list field is not equal");
+        }
+        else if (usingTeradataJdbcDriver(connection)) {
+            assertEqualsIgnoreOrder(queryResult.column(10), column(expectedRows, 10), "list_to_list field is not equal");
```

Comment:
This appears to be differing by `extract(queryResult.column(10))` vs `queryResult.column(10)`. Why is it?

Especially that you didn't have to do this for `row` or `map`.

Reply:
For `row` and for `map` is returned a different type. For `list_to_list` field returned type is `List<PrestoArray>` and expected value had a different type, because `PrestoArray` does not provide public constructor.